utility/subcommands.py

```diff
@@ -0,0 +1,264 @@
+   - The extract_radosgw_admin_commands function:
+     - Identifies radosgw-admin commands within log lines.
+     - Extracts and reconstructs command outputs.
+     - Uses SHA-256 hashes to detect and prevent duplicate entries.
+
+4. **Saving Outputs:**
+   - The save_to_remote function:
+     - Creates the required directory structure.
+     - Saves extracted command outputs into JSON files categorized by subcommands.
+     - Checks for duplicates before appending new entries.
+
+5. **Execution Flow:**
+   - The run function:
+     - Retrieves .log files from the specified directory.
+     - Processes each file, extracting relevant commands and saving outputs.
+
+### **Key Functions:**
+- get_log_files_from_directory(directory): Retrieves all .log files from a directory.
+- process_log_file(file_path, subcomponent_filter, output_directory): Processes each log file for command extraction.
+- extract_radosgw_admin_commands(log_lines): Extracts radosgw-admin commands and reconstructs outputs.
+- compute_output_hash(output): Computes a unique SHA-256 hash for deduplication.
+- save_to_remote(command, output, subcomponent_filter, output_directory): Saves as JSON files.
+- run(log_directory, subcomponent_filter, output_directory): Orchestrates log file processing and output storage.
+
+### **Folder Structure:**
+The script organizes extracted outputs into the specified output directory as follows:
+
+"""
+
+import hashlib
+import json
+import os
+import re
+
+from docopt import docopt
+
+DOC = """
+Standard script to fetch and process log files from a given directory.
+
+Usage:
+    subcommands.py --logdir <log_directory> --filter <subcomponent_filter> --outdir <output_directory>
+    subcommands.py (-h | --help)
+
+Options:
+    -h --help                      Show this help message
+    --logdir <log_directory>       Directory containing log files
+    --filter <subcomponent_filter> Filter logs by subcomponent (e.g., rgw, rbd, rados)
+    --outdir <output_directory>    Directory where output JSON files will be stored
+"""
+
+
+def compute_output_hash(output):
+    """
+    Computes a SHA-256 hash for the given output data.
+    Args:
+        output (dict): The output data to hash.
+    Returns:
+        str: The computed hash value as a hexadecimal string.
+    """
+    return hashlib.sha256(
+        json.dumps(output, sort_keys=True).encode("utf-8")
+    ).hexdigest()
+
+
+def clean_log_line(line):
+    """
+    Cleans log lines by removing timestamps and log level prefixes.
+    Args:
+        line (str): A single line from the log file.
+    Returns:
+        str or None: The cleaned log line or None if empty.
+    """
+    line = re.sub(
+        r"^\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2},\d{3} - cephci - ceph:\d+ - (INFO|DEBUG|ERROR) -\s*",
+        "",
+        line,
+    ).strip()
+    line = re.sub(
+        r"^\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2},\d{3} (INFO|DEBUG|ERROR):?\s*", "", line
+    ).strip()
+    return line if line else None
```

Comment:
I think this would be best done in `ceph-qe-scripts`. Over there, use a log filter to provide a separate artifact during runtime.